File: javaAPI/src/main/java/com/kotlinjavademo/javaAPI/repositories/VehicleRepository.java
package com.kotlinjavademo.javaAPI.repositories;

import com.kotlinjavademo.javaAPI.entities.VehicleEntity;
import org.hibernate.validator.internal.engine.valueextraction.ValueExtractorHelper;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;
import org.springframework.transaction.annotation.Transactional;


@Repository
@Transactional
public interface VehicleRepository extends JpaRepository<VehicleEntity, Long> {
    VehicleEntity getVehicleByRegistration(String registration);
}

File: kotlin/src/main/kotlin/com/kotlinjavademo/kotlin/repositories/VehicleRepository.kt
package com.kotlinjavademo.kotlin.repositories

import com.kotlinjavademo.kotlin.entities.VehicleEntity
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository
import org.springframework.transaction.annotation.Transactional

@Repository
@Transactional
interface VehicleRepository : JpaRepository<VehicleEntity, Long> {
    fun getVehicleByRegistration(registration: String): VehicleEntity?
}